mshv-ioctls/src/ioctls/system.rs

```diff
@@ -0,0 +1,201 @@
+            },
+            flags: creation_flags,
+        };
+        let ret = unsafe { ioctl_with_ref(&self.hv, MSHV_CREATE_PARTITION(), &pr) };
+        if ret >= 0 {
+            // Safe because we verify the value of ret and we are the owners of the fd.
+            let vm_file = unsafe { File::from_raw_fd(ret) };
+            Ok(new_vmfd(vm_file))
+        } else {
+            Err(errno::Error::last())
+        }
+    }
+    ///
+    /// Validate the MSHV version
+    ///
+    pub fn request_version(&self) -> Result<()> {
+        // Safe because we know `self.hv` is a real MSHV fd as this module is the only one that
+        // creates mshv objects.
+        let version: u32 = MSHV_VERSION;
+        let ret = unsafe { ioctl_with_ref(&self.hv, MSHV_REQUEST_VERSION(), &version) };
+        if ret == 0 {
+            Ok(())
+        } else {
+            Err(errno::Error::last())
+        }
+    }
+    ///
+    /// X86 specific call to get list of supported MSRS
+    ///
+    pub fn get_msr_index_list(&self) -> Result<MsrList> {
+        /* return all the MSRs we currently support */
+        Ok(MsrList::from_entries(&[
+            IA32_MSR_TSC,
+            IA32_MSR_EFER,
+            IA32_MSR_KERNEL_GS_BASE,
+            IA32_MSR_APIC_BASE,
+            IA32_MSR_PAT,
+            IA32_MSR_SYSENTER_CS,
+            IA32_MSR_SYSENTER_ESP,
+            IA32_MSR_SYSENTER_EIP,
+            IA32_MSR_STAR,
+            IA32_MSR_LSTAR,
+            IA32_MSR_CSTAR,
+            IA32_MSR_SFMASK,
+            IA32_MSR_MTRR_CAP,
+            IA32_MSR_MTRR_DEF_TYPE,
+            IA32_MSR_MTRR_PHYSBASE0,
+            IA32_MSR_MTRR_PHYSMASK0,
+            IA32_MSR_MTRR_PHYSBASE1,
+            IA32_MSR_MTRR_PHYSMASK1,
+            IA32_MSR_MTRR_PHYSBASE2,
+            IA32_MSR_MTRR_PHYSMASK2,
+            IA32_MSR_MTRR_PHYSBASE3,
+            IA32_MSR_MTRR_PHYSMASK3,
+            IA32_MSR_MTRR_PHYSBASE4,
+            IA32_MSR_MTRR_PHYSMASK4,
+            IA32_MSR_MTRR_PHYSBASE5,
+            IA32_MSR_MTRR_PHYSMASK5,
+            IA32_MSR_MTRR_PHYSBASE6,
+            IA32_MSR_MTRR_PHYSMASK6,
+            IA32_MSR_MTRR_PHYSBASE7,
+            IA32_MSR_MTRR_PHYSMASK7,
+            IA32_MSR_MTRR_FIX64K_00000,
+            IA32_MSR_MTRR_FIX16K_80000,
+            IA32_MSR_MTRR_FIX16K_a0000,
+            IA32_MSR_MTRR_FIX4K_c0000,
+            IA32_MSR_MTRR_FIX4K_c8000,
+            IA32_MSR_MTRR_FIX4K_d0000,
+            IA32_MSR_MTRR_FIX4K_d8000,
+            IA32_MSR_MTRR_FIX4K_e0000,
+            IA32_MSR_MTRR_FIX4K_e8000,
+            IA32_MSR_MTRR_FIX4K_f0000,
+            IA32_MSR_MTRR_FIX4K_f8000,
+            IA32_MSR_TSC_AUX,
+            IA32_MSR_BNDCFGS,
+            IA32_MSR_DEBUG_CTL,
+            IA32_MSR_MISC_ENABLE,
+        ]))
+    }
+}
+#[allow(dead_code)]
+#[cfg(test)]
+mod tests {
+    // Note this useful idiom: importing names from outer (for mod tests) scope.
+    use super::*;
+
+    #[test]
+    fn test_mshv_new() {
```

Comment:
I don't think this test is needed. `Mshv::new()` is covered with most other tests.